### PR TITLE
(maint) Fix issue #98, don't require `pe_compiling_server_aio_build`

### PR DESCRIPTION
**Pull request by MikaelSmith**
Compilation should not require `pe_compiling_server_aio_build`. Skip
calling it if not defined even if `is_pe` is true.

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -77,9 +77,8 @@
 
   # The aio puppet-agent version currently installed on the compiling master
   # (only used in PE)
-  $master_agent_version = $_is_pe ? {
-    true    => pe_compiling_server_aio_build(),
-    default => undef,
+  if ($_is_pe && is_function_available('pe_compiling_server_aio_build')) {
+    $master_agent_version = pe_compiling_server_aio_build()
```
Comment (highb) [inline manifests/params.pp]: Can we add an else to explicitly set it to `undef`? We're going to get warnings all over the place otherwise.

Comment (MikaelSmith) [inline manifests/params.pp]: Ok. I also want to add a spec test.

**Comment by MikaelSmith** [comment]
Updated.

**Comment by highb** [comment]
:+1: once specs pass.

**Comment by MikaelSmith** [comment]
Specs are proving tricky.

**Comment by MikaelSmith** [comment]
I think I got all the tests working. Got a little complicated.

**Comment by highb** [comment]
Heh, that did get a bit complicated. Still, the fundamental change make sense and the specs mostly cover its behavior so :+1:

**Pull request merged**